src/osd/scrubber/pg_scrubber.cc

```diff
@@ -2698,11 +2745,55 @@ void ReplicaReservations::handle_reserve_reject(OpRequestRef op,
   }
 }
 
+void ReplicaReservations::handle_no_reply_timeout()
+{
+  dout(1) << fmt::format(
+	       "{}: timeout! no reply from {}", __func__, m_waited_for_peers)
+	  << dendl;
+
+  m_had_rejections = true;  // preventing any additional notifications
+  send_reject();
+}
+
 std::ostream& ReplicaReservations::gen_prefix(std::ostream& out) const
 {
   return out << m_log_msg_prefix;
 }
 
+ReplicaReservations::no_reply_t::no_reply_t(
+  OSDService* osds,
+  const ConfigProxy& conf,
+  ReplicaReservations& parent,
+  std::string_view log_prfx)
+    : m_osds{osds}
+    , m_conf{conf}
+    , m_parent{parent}
+    , m_log_prfx{log_prfx}
+{
+  using namespace std::chrono;
+  auto now_is = clock::now();
+  milliseconds timeout =
+    milliseconds(conf.get_val<int64_t>("osd_scrub_reservation_timeout"));
```

Comment:
Don't add an event if timeout is set to 0.

Reply:
I've considered this, but preferred to avoid the extra code, as:
- I believe this warning should not be bypassed; and
- One can always assign a large-enough number for the timeout.

Reply:
> I've considered this, but preferred to avoid the extra code, as:
> 
> * I believe this warning should not be bypassed; and
> * One can always assign a large-enough number for the timeout.

we can enforce it by adding a `min`/`max` value to the configuration yaml

Reply:
good idea. Will try that now.

Reply:
Added 'min' values